src/metabase/models/database.clj

```diff
@@ -154,10 +154,14 @@
   "The string to replace passwords with when serializing Databases."
   "**MetabasePass**")
 
-(def ^:const sensitive-fields
-  "List of fields that should be obfuscated in API responses, as they contain sensitive data."
-  [:password :pass :tunnel-pass :tunnel-private-key :tunnel-private-key-passphrase
-   :access-token :refresh-token :service-account-json])
+(defn get-sensitive-fields-for-db
+  "Gets all sensitive fields that should be redacted in API responses for a given database. Calls get-sensitive-fields
+  using the given database's driver, if that driver is valid and registered. Refer to get-sensitive-fields Scaladoc
+  for full details."
+  [database]
+  (if-not (:subprotocol database)
+    driver/default-sensitive-fields ;; preserve existing behavior by returning the default sensitive fields
+    (driver/get-sensitive-fields (driver.u/database->driver database))))
```

Comment:
`database` won't have a `:subprotocol`. Database isn't a JDBC spec, it's a row from the `metabase_database` table in the app DB. The way you did it below with `(driver.u/database->driver database)` is the right way to do it.

Reply:
> * Not all data warehouse DBs we support are JDBC DBs (for example, MongoDB) -- so don't try to use `:subprotocol`. Use `driver.u/database->driver` or just `Database.engine` (these do the same thing) to get the `driver`
> * I don't think this really makes sense as a multimethod, if we can't think of any situation where someone would actually need to override the behavior. I'd rather only add multimethods (and make the driver interface bigger) if we can think of an actual reason someone might possibly need to override it. Since it just derives its values from `connection-properties`, a driver author should be marking things as sensitive _there_. It could be confusing if it seemed like there were two places they needed to do it. I'd make this a utility function instead
> 
> Besides those minor code organization suggestions this looks good

All of this should be addressed now (no more multimethod, just a single utility fn).